### PR TITLE
Add Crypto.org support

### LINKED ISSUE #1638: Add Osmosis/CryptoOrg support

### PR DESCRIPTION
**Pull request by catenocrypt**
## Description

Add support for Crypto.org chain, CRO.
Based on Cosmos; Address, proto & Signer is reused.
Derivation path is `"m/44'/394'/0'/0/0"`
Partly fixes #1638 .

## Testing instructions

Unit tests, Cryptoorg.

## Types of changes

* New feature (non-breaking change which adds functionality)

## Checklist

- [x] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain
- [x] If there is a related Issue, mention it in the description (e.g. Fixes #<issue_number> ).

### CODE REVIEW

**Review by vikmeup — APPROVED** [review]

**Comment by hewigovens** [comment]
Does CRO have staking?

File: registry.json [hunk 1]
```diff
@@ -1741,5 +1741,30 @@
...
     }
+  },
+  {
+    "id": "cryptoorg",
+    "name": "Crypto.org",
+    "coinId": 394,
+    "symbol": "CRO",
+    "decimals": 8,
+    "blockchain": "Cosmos",
+    "derivationPath": "m/44'/394'/0'/0/0",
```
Comment (hewigovens) [inline registry.json:1752]: can you cross check with keplr extension wallet? should be compatible

Comment (catenocrypt) [reply registry.json:1752]: Yes, confirmed, Kepl Chrome extension wallet results in the same addresses.

**Comment by catenocrypt** [comment]
> Does CRO have staking?

Haven't seen anything on staking.

File: src/interface/TWAnyAddress.cpp [hunk 2]
```diff
@@ -90,6 +90,16 @@ TWData* _Nonnull TWAnyAddressData(struct TWAnyAddress* _Nonnull address) {
         break;
     }
 
+    case TWCoinTypeCryptoorg:
```
Comment (hewigovens) [inline src/interface/TWAnyAddress.cpp:93]: merge this into Cosmos case

File: registry.json [hunk 3]
```diff
@@ -1741,5 +1741,30 @@
       "clientPublic": "https://rpc.ftm.tools",
       "clientDocs": "https://eth.wiki/json-rpc/API"
     }
+  },
+  {
+    "id": "cryptoorg",
+    "name": "Crypto.org",
```
Comment (hewigovens) [inline registry.json:1747]: the generate type name is `TWCoinTypeCryptoorg`, can we do `TWCoinTypeCryptoOrg`? add display name to `Crypto.org`

**Review by hewigovens — APPROVED** [review]

File: src/interface/TWAnyAddress.cpp (hunk 2 above)
Comment (catenocrypt) [reply src/interface/TWAnyAddress.cpp:93]: Good point, leftover from when Address class was duplicated.

**Pull request merged**